cmd/zc_traverser_blob.go

```diff
@@ -162,18 +164,33 @@ func (t *blobTraverser) Traverse(preprocessor objectMorpher, processor objectPro
 	blobProperties, isBlob, isDirStub, propErr := t.getPropertiesIfSingleBlob()
 
 	if stgErr, ok := propErr.(azblob.StorageError); ok {
+
 		// Don't error out unless it's a CPK error just yet
 		// If it's a CPK error, we know it's a single blob and that we can't get the properties on it anyway.
 		if stgErr.ServiceCode() == common.CPK_ERROR_SERVICE_CODE {
-			return errors.New("this blob uses customer provided encryption keys (CPK). At the moment, AzCopy does not support CPK-encrypted blobs. " +
-				"If you wish to make use of this blob, we recommend using one of the Azure Storage SDKs")
+			if t.getProperties {
+				return errors.New("this blob uses customer provided encryption keys (CPK). At the moment, AzCopy does not support CPK-encrypted blobs. " +
+					"If you wish to make use of this blob, we recommend using one of the Azure Storage SDKs")
+			} else {
+				blobProperties = &azblob.BlobGetPropertiesResponse{}
+				isBlob = true
+				isDirStub = true
```

Comment:
the only problem here is we have to assume root is a directory to allow further recursive listing.

Reply:
We could read the recursive flag and if that's false assume a file though

Reply:
We could just skip the getProperty check and blindly add the empty blob properties.

Reply:
Also this method is actually Traverse not IsDirectory. I believe IsDirectory is only called for copy and sync

Reply:
I'm still confused on why we don't throw on CPK error immediately in the case of !t.getProperties?

Reply:
we don't need CPK to delete a file, so even if we failed to get properties we can still move on and pretend like everything is ok 
https://learn.microsoft.com/en-us/rest/api/storageservices/delete-blob

Reply:
Ah ok I actually thing that would be very useful to add in as an inline comment